src/main/java/org/broadinstitute/hellbender/utils/pairhmm/DragstrLocus.java

```diff
@@ -0,0 +1,218 @@
+package org.broadinstitute.hellbender.utils.pairhmm;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.broadinstitute.hellbender.utils.Nucleotide;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+
+import java.io.*;
+import java.util.Arrays;
+
+public class DragstrLocus {
+
+    private final int chromosomeIndex;
+    private final long start;
+    private final int repeats;
+    private final byte[] unit;
+
+    private DragstrLocus(final int chrIdx, final long start, final byte[] unit, final int repeats) {
+        chromosomeIndex = chrIdx;
+        this.start = start;
+        this.unit = unit;
+        this.repeats = repeats;
+    }
+
+    public static DragstrLocus make(final int chrIdx, final long start, final byte[] unit, final int repeatCount) {
+        ParamUtils.isPositiveOrZero(chrIdx, "chromosome index");
+        ParamUtils.isPositive(start, "start position");
+        final byte[] unitCloned = Utils.nonNull(unit).clone();
+        for (final byte base : unit) {
+            if (!Nucleotide.decode(base).isStandard()) {
+                throw new IllegalArgumentException("bad bases in " + new String(unit));
+            }
+        }
+        ParamUtils.isPositive(repeatCount, "repeat count");
+        return new DragstrLocus(chrIdx, start, unit.clone(), repeatCount);
+    }
+
+    public long getStart() {
+        return start;
+    }
+
+    public long getEnd() {
+        return start + unit.length * repeats - 1;
+    }
+
+    public int getPeriod() {
+        return unit.length;
+    }
+
+    byte[] getUnitUnsafe() {
+        return unit;
+    }
+
+    public byte[] getUnit() {
+        return unit.clone();
+    }
+
+    public int getRepeats() {
+        return repeats;
+    }
+
+    public static BinaryTableWriter<DragstrLocus> binaryWriter(final OutputStream out, final String path) {
```

Comment:
Perhaps we should declare the DragstrLocus as a feature for the feature reader? that means these methods can all be pushed in the feature codec machinery and consequently it should cut down on the amount of code added here..